lib/util/git/index.ts

```diff
@@ -673,19 +673,44 @@ export async function isBranchModified(branchName: string): Promise<boolean> {
     }
     logger.warn({ err }, 'Error checking last author for isBranchModified');
   }
-  const { gitAuthorEmail } = config;
-  if (
-    lastAuthor === gitAuthorEmail ||
-    config.ignoredAuthors.some((ignoredAuthor) => lastAuthor === ignoredAuthor)
-  ) {
-    // author matches - branch has not been modified
-    logger.debug('branch.isModified() = false');
+  const { gitAuthorEmail, ignoredAuthors } = config;
+
+  const includedAuthors = new Set(committedAuthors);
+
+  if (gitAuthorEmail) {
+    includedAuthors.delete(gitAuthorEmail);
+  }
+
+  for (const ignoredAuthor of ignoredAuthors) {
+    includedAuthors.delete(ignoredAuthor);
+  }
+
+  if (includedAuthors.size === 0) {
+    // authors all match - branch has not been modified
+    logger.debug(
+      {
+        branchName,
+        baseBranch,
+        committedAuthors: [...committedAuthors],
+        includedAuthors: [...includedAuthors],
+        gitAuthorEmail,
+        ignoredAuthors,
+      },
+      'branch.isModified() = false',
+    );
```

Comment:
This is a pretty long debug message to print every time a branch is unmodified. I would prefer that the metadata is in a trace log, or that the debug is greatly shortened